Modules/_ctypes/_ctypes.c

```diff
@@ -2731,11 +2731,16 @@ PyCData_reduce(PyObject *myself, PyObject *args)
                         "ctypes objects containing pointers cannot be pickled");
         return NULL;
     }
-    return Py_BuildValue("O(O(NN))",
-                         _unpickle,
-                         Py_TYPE(myself),
-                         PyObject_GetAttrString(myself, "__dict__"),
-                         PyBytes_FromStringAndSize(self->b_ptr, self->b_size));
+    PyObject *dict = PyObject_GetAttrString(myself, "__dict__");
+    if (dict == NULL) {
+        return NULL;
+    }
+    PyObject *bytes = PyBytes_FromStringAndSize(self->b_ptr, self->b_size);
```

Comment:
It is enough to check for error only `dict`. `PyBytes_FromStringAndSize()` can be passed directly to `Py_BuildValue()`.

Reply:
You could even use `y#` instead of `PyBytes_FromStringAndSize()`.

Reply:
Thank you, Serhiy. I've updated the PR.